packages/angular/build/src/builders/application/execute-build.ts

```diff
@@ -160,6 +169,15 @@ export async function executeBuild(
     executionResult.htmlBaseHref = options.baseHref;
   }
 
+  // Create server app engine manifest
+  if (serverEntryPoint) {
+    executionResult.addOutputFile(
+      SERVER_APP_ENGINE_MANIFEST_FILENAME,
+      generateAngularServerAppEngineManifest(i18nOptions, baseHref, undefined),
+      BuildOutputFileType.SSRServer,
+    );
```

Comment:
This appears to be done twice in this file and once in the post build actions?  Can this be done once at the end?

Reply:
This process is required in two stages. Initially, we create a partial manifest containing only the essential information needed to run the app, and extract routes at this stage. Once the routes have been extracted, we regenerate the manifest to include these routes, reducing the need to do this during the first request. This approach helps minimize overhead, particularly in serverless environments.

Reply:
This process must be completed in two stages:

**Stage 1**: Generate a minimal manifest containing only the necessary information to run the app.

**Stage 2**: Regenerate the manifest, incorporating all the information discovered during the app's bootstrapping phase. Such as headers for each route.

We handle this during the build process to avoid the overhead of route discovery on each request, which can become a bottleneck in serverless environments.